### PR TITLE
DataBaseService_EditUserLogic

### PR DESCRIPTION
**Pull request by strannick-rodion**

### CODE REVIEW

File: Services/DataBaseService/BrokerConsumers/EditUserConsumer.cs [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+
+        public EditUserConsumer([FromServices] IUserRepository userRepository)
+        {
+            this.userRepository = userRepository;
+        }
+
+        private OperationResult EditUser(InternalEditUserInfoRequest request)
+        {
+            userRepository.EditUser(request.User,request.UserPasswords);
+
```
Comment (nmamro) [inline Services/DataBaseService/BrokerConsumers/EditUserConsumer.cs]: Extra empty line :)

File: Services/DataBaseService/Repositories/Interfaces/IUserRepository.cs [hunk 2]
```diff
@@ -1,4 +1,5 @@
-﻿using DTO;
+﻿using DataBaseService.Database.Models;
```
Comment (nmamro) [inline Services/DataBaseService/Repositories/Interfaces/IUserRepository.cs]: Do we use this one?

File: Services/DataBaseService/Repositories/Interfaces/IUserRepository.cs [hunk 3]
```diff
@@ -14,5 +15,6 @@ public interface IUserRepository
         User GetUserById(Guid userId);
 
         void DeleteUser(Guid userIdCredential);
+        void EditUser(User user, PasswordHashChangeRequest password);
```
Comment (nmamro) [inline Services/DataBaseService/Repositories/Interfaces/IUserRepository.cs:18]: No empty line this time =)

File: Services/DataBaseService/Repositories/UserRepository.cs [hunk 4]
```diff
@@ -67,5 +67,28 @@ public void DeleteUser(Guid userId)
             dbUserCredential.IsActive = false;
             dbContext.SaveChanges();
         }
+
+        public void EditUser(User user, PasswordHashChangeRequest password)
+        {
+            var dbUser = dbContext.Users.FirstOrDefault(uc => uc.Id == user.Id);
+            if (dbUser == null)
```
Comment (nmamro) [inline Services/DataBaseService/Repositories/UserRepository.cs]: `==` or `!=`?

File: Services/DataBaseService/Repositories/UserRepository.cs [hunk 5]
```diff
@@ -67,5 +67,28 @@ public void DeleteUser(Guid userId)
...
+                dbUser.LastName = user.LastName;
+                dbUser.FirstName = user.FirstName;
+                dbUser.Birthday = user.Birthday;
+                dbContext.SaveChanges();
+
+                if (password != null)
+                { 
+                    var dbUserCredential = dbContext.UsersCredentials.FirstOrDefault(uc => uc.UserId == user.Id);
+
+                    if (dbUserCredential != null)   dbUserCredential.PasswordHash = password.NewPasswordHash;
```
Comment (nmamro) [inline Services/DataBaseService/Repositories/UserRepository.cs]: 1) `if` style guide?
2) Shouldn't we `SaveChanges()`?

**Review by nmamro — CHANGES_REQUESTED** [review]

File: Services/DataBaseService/BrokerConsumers/EditUserConsumer.cs (hunk 1 above)
Comment (strannick-rodion) [reply Services/DataBaseService/BrokerConsumers/EditUserConsumer.cs]: fix

File: Services/DataBaseService/Repositories/Interfaces/IUserRepository.cs (hunk 2 above)
Comment (strannick-rodion) [reply Services/DataBaseService/Repositories/Interfaces/IUserRepository.cs]: fix

File: Services/DataBaseService/Repositories/UserRepository.cs (hunk 4 above)
Comment (strannick-rodion) [reply Services/DataBaseService/Repositories/UserRepository.cs]: fix

File: Services/DataBaseService/Repositories/UserRepository.cs (hunk 5 above)
Comment (strannick-rodion) [reply Services/DataBaseService/Repositories/UserRepository.cs]: 2.fix

Comment (strannick-rodion) [reply Services/DataBaseService/Repositories/UserRepository.cs]: 1 fix

File: Services/DataBaseService/Repositories/Interfaces/IUserRepository.cs (hunk 3 above)
Comment (strannick-rodion) [reply Services/DataBaseService/Repositories/Interfaces/IUserRepository.cs:18]: fix

**Review by Cerberous007 — APPROVED** [review]

**Review by nmamro — APPROVED** [review]

**Review by LK777LK — APPROVED** [review]

**Review by denis-kriv — APPROVED** [review]

**Pull request merged**